comp/metadata/clusteragent/impl/cluster_agent.go

```diff
@@ -0,0 +1,198 @@
+		Timestamp:   time.Now().UnixNano(),
+		Metadata:    dca.getMetadata(),
+		UUID:        uuid.GetUUID(),
+	}
+}
+
+func (dca *datadogclusteragent) initMetadata(metadata map[string]interface{}) {
+	tool := "undefined"
+	toolVersion := ""
+	installerVersion := ""
+
+	install, err := installinfo.Get(dca.conf)
+	if err == nil {
+		tool = install.Tool
+		toolVersion = install.ToolVersion
+		installerVersion = install.InstallerVersion
+	}
+	metadata["cluster_id_error"] = dca.clusteridErr
+	metadata["install_method_tool"] = tool
+	metadata["install_method_tool_version"] = toolVersion
+	metadata["install_method_installer_version"] = installerVersion
+	metadata["agent_version"] = version.AgentVersion
+	metadata["agent_startup_time_ms"] = pkgconfigsetup.StartTime.UnixMilli()
+	metadata["flavor"] = flavor.GetFlavor()
+}
+
+func (dca *datadogclusteragent) getAadmissionControllerConfig(metadata map[string]interface{}) {
+	metadata["feature_admission_controller_enabled"] = dca.conf.GetBool("admission_controller.enabled")
+	metadata["feature_admission_controller_inject_config_enabled"] = dca.conf.GetBool("admission_controller.inject_config.enabled")
+	metadata["feature_admission_controller_inject_tags_enabled"] = dca.conf.GetBool("admission_controller.inject_tags.enabled")
+	metadata["feature_apm_config_instrumentation_enabled"] = dca.conf.GetBool("apm_config.instrumentation.enabled")
+	metadata["feature_admission_controller_validation_enabled"] = dca.conf.GetBool("admission_controller.validation.enabled")
+	metadata["feature_admission_controller_mutation_enabled"] = dca.conf.GetBool("admission_controller.mutation.enabled")
+	metadata["feature_admission_controller_auto_instrumentation_enabled"] = dca.conf.GetBool("admission_controller.auto_instrumentation.enabled")
+	metadata["feature_admission_controller_cws_instrumentation_enabled"] = dca.conf.GetBool("admission_controller.cws_instrumentation.enabled")
+	metadata["feature_cluster_checks_enabled"] = dca.conf.GetBool("cluster_checks.enabled")
+	metadata["feature_autoscaling_workload_enabled"] = dca.conf.GetBool("autoscaling.workload.enabled")
+	metadata["feature_external_metrics_provider_enabled"] = dca.conf.GetBool("external_metrics_provider.enabled")
+	metadata["feature_external_metrics_provider_use_datadogmetric_crd"] = dca.conf.GetBool("external_metrics_provider.use_datadogmetric_crd")
+	metadata["feature_compliance_config_enabled"] = dca.conf.GetBool("compliance_config.enabled")
+}
+
+func (dca *datadogclusteragent) getMetadata() map[string]interface{} {
+	metadata := map[string]interface{}{}
+	dca.initMetadata(metadata)
+	metadata["leader_election"] = dca.conf.GetBool("leader_election")
+	metadata["is_leader"] = false
+	if dca.conf.GetBool("leader_election") {
+		if leaderEngine, err := leaderelection.GetLeaderEngine(); err == nil {
+			metadata["is_leader"] = leaderEngine.IsLeader()
+		}
+	}
+	if str, err := fetchDatadogClusterAgentConfig(dca.conf); err == nil {
+		metadata["full_configuration"] = str
```

Comment:
Please leverage `inventories_configuration_enabled` settings to know if we should include any configuration in our paylaods.